cpp/oneapi/dal/table/detail/csr.hpp

```diff
@@ -0,0 +1,196 @@
+    ///
+    /// @param data           The pointer to values in the CSR layout.
+    /// @param column_indices The pointer to column indices in the CSR layout.
+    /// @param row_indices    The pointer to row indices in the CSR layout.
+    /// @param row_count      The number of rows in the corresponding dense table.
+    /// @param column_count   The number of columns in the corresponding dense table.
+    /// @param data_deleter   The deleter that is called on the ``data`` when the last table that refers it
+    ///                       is out of the scope.
+    /// @param column_deleter The deleter that is called on the ``column_indices`` when the last table that refers it
+    ///                       is out of the scope.
+    /// @param row_deleter    The deleter that is called on the ``row_indices`` when the last table that refers it
+    ///                       is out of the scope.
+    /// @param indexing       The indexing scheme used to access data in the CSR layout. Support only :literal:`csr_indexing::one_based`.
+    template <typename Data, typename DataDeleter, typename ColumnDeleter, typename RowDeleter>
+    csr_table(const Data* data,
+              const std::int64_t* column_indices,
+              const std::int64_t* row_indices,
+              std::int64_t row_count,
+              std::int64_t column_count,
+              DataDeleter&& data_deleter,
+              ColumnDeleter&& column_deleter,
+              RowDeleter&& row_deleter,
+              csr_indexing indexing = csr_indexing::one_based) {
+        init_impl(detail::default_host_policy{},
+                  column_count,
+                  row_count,
+                  data,
+                  column_indices,
+                  row_indices,
+                  std::forward<DataDeleter>(data_deleter),
+                  std::forward<ColumnDeleter>(column_deleter),
+                  std::forward<RowDeleter>(row_deleter),
+                  indexing);
+    }
+
+    /// The unique id of the csr table type.
+    std::int64_t get_kind() const {
+        return kind();
+    }
+
+    /// Returns the :literal:`data` pointer cast to the :literal:`Data` type. No checks are
+    /// performed that this type is the actual type of the data within the table.
+    template <typename Data>
+    const Data* get_data() const {
+        return reinterpret_cast<const Data*>(this->get_data());
+    }
+
+    /// The pointer to the data block within the table.
+    /// Should be equal to ``nullptr`` when :expr:`row_count == 0` and :expr:`column_count == 0`.
+    const void* get_data() const;
+
+    /// Returns the :literal:`column_indices` pointer.
+    const std::int64_t* get_column_indices() const;
+
+    /// Returns the :literal:`row_indices` pointer.
+    const std::int64_t* get_row_indices() const;
+
+private:
+    explicit csr_table(detail::csr_table_iface* impl) : table(impl) {}
+
+    bool correct_indices(const std::int64_t size,
```

Comment:
check_indices + throw an exception instead of returning boolean flag

Reply:
It was fixed